### PR TITLE
8293253: [lw4] Code fixing access flags in old class files is incorrect and incomplete

### PR DESCRIPTION
**Pull request by fparain**
Please review this code fixing the code that adds the missing ACC_SUPER/ACC_IDENTITY flag in old class files.
Those changes fix both JDK-8291719 and JDK-8293169.

Tested with Mach5, tier1-3.

But those changes also trigger new failures that should be addressed in additional CR:

java/lang/invoke/defineHiddenClass/BasicTest.java:
  - assertion failure because test doesn't expect ACC_IDENTITY to be present
    not sure about the fix if non-Valhalla configurations are run
    BTW, the test already uses ACC_IDENTITY in some places

 java/lang/reflect/AccessFlag/ClassAccessFlagTest.java:
  - another Java test surprised by the presence of ACC_IDENTITY

serviceability/jvmti/RedefineClasses/RedefineObject.java:
  - changes in access flags leads to the class redefinition code to reject
    an attempt to redefine a class because it sees a mismatch in access flags
    (forbidden by class redefinition rules)

tools/javac/preview/PreviewTest.java
  - not sure about this one, need input from a javac expert

Still some failures in JCK test, vm/classfmt/cpl and vm/instr/invokespecial
but they will be addressed in another patch.

Thank you,

Fred

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8293253](https://bugs.openjdk.org/browse/JDK-8293253): [lw4] Code fixing access flags in old class files is incorrect and incomplete

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - Committer) ⚠️ Review applies to [41aa059a](https://git.openjdk.org/valhalla/pull/739/files/41aa059a151be37748b1a0315de985e4ee969d65)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla pull/739/head:pull/739` \
`$ git checkout pull/739`

Update a local copy of the PR: \
`$ git checkout pull/739` \
`$ git pull https://git.openjdk.org/valhalla pull/739/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 739`

View PR using the GUI difftool: \
`$ git pr show -t 739`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/739.diff">https://git.openjdk.org/valhalla/pull/739.diff</a>

</details>

### CODE REVIEW

**Comment by MrSimms** [comment]
Looks good.

There is further clean up to be had, but I think that is out of scope this bug (fixing test issues).

Thanks Frederic !

File: src/hotspot/share/classfile/classFileParser.cpp [hunk 1]
```diff
@@ -3338,6 +3338,16 @@ u2 ClassFileParser::parse_classfile_inner_classes_attribute(const ClassFileStrea
       flags |= JVM_ACC_ABSTRACT;
     }
 
+    if (EnableValhalla) {
+      if(!supports_inline_types()) {
```
Comment (TobiHartmann) [inline src/hotspot/share/classfile/classFileParser.cpp:3342]:
```suggestion
      if (!supports_inline_types()) {
```

**Review by TobiHartmann — APPROVED** [review]
Looks good.

**Comment by fparain** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**